### PR TITLE
Feature/alg 558 ssd embed anchors

### PR DESCRIPTION
**Pull request by yurkovak**
It's safer to store anchors inside of a checkpoint and more correct to convert preds to boxes and scores inside of a mode

### CODE REVIEW

File: src/super_gradients/training/models/detection_models/ssd.py
Comment (ofrimasad) [inline src/super_gradients/training/models/detection_models/ssd.py:164]: remove comment

**Review by ofrimasad — COMMENTED** [review]
looks like we are getting awfully close to using utils.detection_utils.Anchors.
Can we move to the new Anchors maybe?

File: src/super_gradients/training/models/detection_models/ssd.py
Comment (yurkovak) [reply src/super_gradients/training/models/detection_models/ssd.py]: Oh, right, thanks
801f61e

**Comment by yurkovak** [comment]
On hold, waiting for some fixes yamls to arrive from master. Want to see everything training with the proposed changes before we merge it, but for now it cannot be trained even in master

File: src/super_gradients/training/models/detection_models/ssd.py
Comment (shaydeci) [inline src/super_gradients/training/models/detection_models/ssd.py:162]: This means that we aren't able to visualise training predictions then...If this takes a lot of time we can just drop it, but otherwise maybe we should reconsider.

File: src/super_gradients/training/models/detection_models/ssd.py
Comment (shaydeci) [inline src/super_gradients/training/models/detection_models/ssd.py:178]: Redundant parenthesis.

**Review by shaydeci — CHANGES_REQUESTED** [review]

**Review by shaydeci — APPROVED** [review]
LGTM

**Pull request merged**
